### DISCUSSION TITLE
Power fluctuations zeroed many Tasmota devices - The world stopped working

### DISCUSSION THREAD

**Discussion by brivett** [arendst/Tasmota#16967] [Q&A] [answered]
Hi all

I have 40+ Tasmota devices around my property. Mainly Sonoffs (Including Minis, TH10's and two Zigbee Bridge Pros)  Other vendor devices include Athom boards and sockets and generic IR transmitters.

Yesterday, we suffered a power cut.  Whilst restoring the power, the power company bounced the supply causing a short burst of on-off activity.

Several Tasmota devices reset themselves to factory defaults.  My phone was swamped with Tasmota Wifi points....  Particularly annoying were the Zigbee Bridge Pros running Zigbee2Tasmota, which lost their settings and pairing for 40 zigbee devices.    the pure tasmota devices could be restored from backups of their configs - The Zigbee2Tasmota devices do not make that possible.  Because I had to rebuild my zigbee world, I chose this moment to switch to Zigbee2MQTT - I can back that up easily. (and have)

I believe this reset behaviour is part of the design, but I really question why...  knowing this behaviour is there and cannot (currently) be disabled I will need to consider a migration as I can't accept the risk of this again (or at least the resultant service loss)

Are there any plans to make this behaviour a configurable option ?

Thanks
Barry

**Comment by pkkrusty** [comment, marked as answer]
It is configurable with setoption65. Sorry you learned the hard way.

**Reply by s-hadinger** [reply]
For more details see https://tasmota.github.io/docs/Device-Recovery/#recovery-techniques

Indeed you always learn this after the problem occurs. But be aware that disabling this removes the last chance of recovery for a bricked device. Preferably do it on device you can serial flash

**Comment by barbudor** [comment]
I am not disabling this feature and have a strategy to automatically recover reseted devices

1. All devices configuration are backup nighly to a DMP file which is named by the MAC address of the device (123456789012.dmp). This is saved to a private github repo for archive and change tracking. The last version of the DMP file is accessible through a HTTP (not S) local web server
2. All devices run a custom build firmware which includes:
- Wifi and MQTT settings for immediate reconnection 
- A auto-enabled rule `on wifi#connected do WebGetConfig http://myserver/tasmota-conf/%id.dmp endon`

So that in any case of reset, the device automatically reload it last-known-good configuration.
As the loaded configuration do not include the Rule itself, restoring the last-known-good configuration replace the WebGetConfig rule from the firmware.

My user_config_override.h details for that (on top of SSID and MQTT):
```
#define USER_RULE1        "on wifi#connected do WebGetConfig http://myserver/tasmota-conf/%id%.dmp endon"
#define USER_BACKLOG      "Rule1 1"
#define USE_WEBGETCONFIG
```

**Reply by s-hadinger** [reply]
Wow!!!

**Reply by Jason2866** [reply]
Thx, for sharing this clever backup & auto restore solution!

**Comment by brivett** [comment]
Thank you pkkrusty and s-hadinger

I wish I'd found the option 65 earlier....  pain is learning, every day is a school day...

For me, the cost of a bricked device is the $15 device, the cost of rebuilding is much, much greater,  Plus I've been gradually replacing some of the early Sonoffs with nice devices from other manufacturers.  I actually binned a couple today rather than reconfigure.

I do like barbudors approach, however I chose not to run any customised firmware... Thank you for detailing it.

Looks like I need to work around my collection with option 65    

I am SOOO pleased that I can make this change and stick with Tasmota  :-)

Cheers
Barry

**Comment by brivett** [comment]
So I've made the switch to so65 1 on 43 devices...  Most are on v10.0.0

Strangely I found one device (a Wemos Node) running 10.0.0(Sensors) did not recognise option 65

Is it removed from the sensors version ?

**Reply by barbudor** [reply]
Nope
Available in all builds

May be a typo when typing in the command for this one ?

BTW, even if you don't want to go for my recovery scheme, saving - at least from time to time - the configuration file can speed up such reconfiguration nightmare
